soco/discovery.py

```diff
@@ -54,6 +56,23 @@ def discover(timeout=1, include_invisible=False, interface_addr=None):
 
     """
 
+    def create_socket(interface_addr=None):
+        """ A helper function for creating a socket for discover purposes.
+
+        Create and return a socket with appropriate options set for multicast.
+        """
+
+        _sock = socket.socket(
+            socket.AF_INET, socket.SOCK_DGRAM, socket.IPPROTO_UDP)
+        # UPnP v1.0 requires a TTL of 4
+        _sock.setsockopt(socket.IPPROTO_IP, socket.IP_MULTICAST_TTL,
```

Comment:
You call _sock.setsockopt twice if the interface_addr is set - an if/else would only set it once.